-1 
Tested at: f867e11eed8390b3c921fcca7579554dcfa47b97
 I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 32 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-06-09-2300/src/DetectorDescription/Core/src/Sphere.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-06-09-2300/src/DetectorDescription/Core/src/StoresInstantiation.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-06-09-2300/src/DetectorDescription/Core/src/Torus.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-06-09-2300/src/DetectorDescription/Core/src/Trap.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-06-09-2300/src/DetectorDescription/Core/src/TruncTubs.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-06-09-2300/src/DetectorDescription/Core/src/DDSpecifics.cc:86:44: error: unused variable 'it' [-Werror,-Wunused-variable]
  std::vector<std::string>::const_iterator it = v.begin();
                                           ^
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-06-09-2300/src/DetectorDescription/Core/src/Tubs.cc 
1 error generated.
Entering library rule at DetectorDescription/OfflineDBLoader
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9535/8123/summary.html

--- a/DetectorDescription/Core/src/DDSpecifics.cc
+++ b/DetectorDescription/Core/src/DDSpecifics.cc
@@ -83,7 +83,6 @@ std::ostream & operator<<( std::ostream  & os, const DDSpecifics & sp)
 
 std::ostream & operator<<( std::ostream & os, const std::vector<std::string> & v)
 {
-  std::vector<std::string>::const_iterator it = v.begin();
   for( const auto& it : v ) {
     os << it << std::endl;
   }

